src/bin/pg_autoctl/supervisor.c

```diff
@@ -153,6 +424,14 @@ supervisor_start(Service services[], int serviceCount, const char *pidfile)
 		return false;
 	}
 
+	/*
+	 * Let the dynamic handler decide about dynamic services
+	 */
+	if (dynamicHandler)
+	{
+		dynamicHandler(&supervisor, dynamicHandlerArg);
+	}
+
```

Comment:
I still don't understand why we need an indirection here. It needs to be either better commented, or just gotten rid of, if that's possible to implement directly in the main loop.

Reply:
Same as above. Since this is cut from the bigger PR, the actual usage is not visible.
See here https://github.com/citusdata/pg_auto_failover/pull/539 the complete code with the usage.